src/app.ts

```diff
@@ -277,6 +282,61 @@ app.use(async (ctx, next) => {
     await next();
 });
 
+app.use(async (ctx, next) => {
+    const request = ctx.req;
+    const host = request.header('host');
+    if (!host) {
+        // TODO handle
+        throw new Error('No Host header');
+    }
+    ctx.set('role', 'Anonymous');
+
+    const authorization = request.header('authorization');
+
+    if (!authorization) {
+        return next();
+    }
+
+    const [match, token] = authorization.match(/Bearer\s+(.*)$/) || [null];
+
+    if (!match) {
+        throw new Error('Invalid Authorization header');
+    }
+
+    let protocol = 'https';
+    // We allow insecure requests when not in production for things like testing
+    if (process.env.NODE_ENV !== 'production' && !request.raw.url.startsWith('https')) {
```

Comment:
Should the 2nd conditional be IF starts with `https`

Reply:
Nope - we're allowing the protocol to be `http` but only if we're not in production - otherwise we force to `https`